tools/testrunner/junit_test.go

```diff
@@ -115,6 +115,46 @@ func TestMergeReports_MissingRerun(t *testing.T) {
 	require.Equal(t, errors.New("expected rerun of all failures from previous attempt, missing: [TestCallbacksSuite/TestWorkflowCallbacks_InvalidArgument]"), report.reportingErrs[1])
 }
 
+func TestAppendAlertsSuite(t *testing.T) {
+	j := &junitReport{}
+	alerts := []alert{
+		{Kind: alertKindDataRace, Summary: "Data race detected", Details: "WARNING: DATA RACE\n...", Tests: []string{"go.temporal.io/server/tools/testrunner.TestShowPanic"}},
+		{Kind: alertKindPanic, Summary: "This is a panic", Details: "panic: This is a panic\n...", Tests: []string{"TestPanicExample"}},
+	}
+	j.appendAlertsSuite(alerts)
+
+	require.Len(t, j.Suites, 1)
+	suite := j.Suites[0]
+	require.Equal(t, "ALERTS", suite.Name)
+	require.Equal(t, 2, suite.Failures)
+	require.Equal(t, 2, suite.Tests)
+	require.Len(t, suite.Testcases, 2)
+
+	// Validate the first testcase looks like a DATA RACE alert and includes test in name.
+	tc0 := suite.Testcases[0]
+	require.Contains(t, tc0.Name, "DATA RACE")
+	require.Contains(t, tc0.Name, "TestShowPanic")
+	require.NotNil(t, tc0.Failure)
+	require.Equal(t, "DATA RACE", tc0.Failure.Message)
+	require.Contains(t, tc0.Failure.Data, "WARNING: DATA RACE")
+	require.Contains(t, tc0.Failure.Data, "Detected in tests:")
+	require.Contains(t, tc0.Failure.Data, "go.temporal.io/server/tools/testrunner.TestShowPanic")
+
+	// Validate the second testcase looks like a PANIC alert and includes test in name.
+	tc1 := suite.Testcases[1]
+	require.Contains(t, tc1.Name, "PANIC")
+	require.Contains(t, tc1.Name, "TestPanicExample")
+	require.NotNil(t, tc1.Failure)
+	require.Equal(t, "PANIC", tc1.Failure.Message)
+	require.Contains(t, tc1.Failure.Data, "panic: This is a panic")
+	require.Contains(t, tc1.Failure.Data, "Detected in tests:")
+	require.Contains(t, tc1.Failure.Data, "TestPanicExample")
+
+	// Ensure totals updated at the top level.
+	require.Equal(t, 2, j.Failures)
+	require.Equal(t, 2, j.Tests)
+}
```

Comment:
I wonder if it would be easier to leverage `requireReportEquals` to just compare the file output verbatim?

(`TestMergeReports_MultipleReports` is not a good example, it should probably also do that, too)

Reply:
Why's that? If you compare it with an empty file at first, you then copy the expected file in there and the next run should work, no? Assuming the output is deterministic (I'd expect it to).